worker/runcworker/worker.go

```diff
@@ -104,22 +107,53 @@ func (w *runcworker) Exec(ctx context.Context, meta worker.Meta, root cache.Moun
 		return err
 	}
 
-	logrus.Debugf("> running %s %v", id, meta.Args)
+	nullIO, err := runc.NewNullIO()
+	if err != nil {
+		return errors.Wrap(err, "creating new NULL IO")
+	}
 
-	status, err := w.runc.Run(ctx, id, bundle, &runc.CreateOpts{
-		IO: &forwardIO{stdin: stdin, stdout: stdout, stderr: stderr},
+	logrus.Debugf("> creating %s %v", id, meta.Args)
+	err = w.runc.Create(ctx, id, bundle, &runc.CreateOpts{
+		IO: nullIO,
```

Comment:
Do you have trouble getting the create/start logic with streams to work? Seems there might be an issue with controlling the wait return in `go-runc`.

Reply:
Yes, The problem is I am not able to find clean way to wait till task gets finished.
Waiting on PID or stream output.. nothing worked so far.

The initial approach, where clean way to wait for container exits, has issue of waiting till PID gets created,  ref : https://github.com/moby/buildkit/pull/174/commits/caed56f64662f4a6b2db2086be8b085b36c40028#diff-d85f89881dedb8116f6ed131b87de008R111

Reply:
https://github.com/tonistiigi/buildkit/commit/aee146866792463afc4c04b5f4ef489b84cdf7a5 is an example how to work around it. I created an issue in https://github.com/containerd/go-runc/issues/31 as well. Let me know if you have any questions.

We could also consider using `runc exec` for actually running the user process as this is getting quite hacky.